### PR TITLE
Improvements to OCKHeartView

### PR DESCRIPTION
**Pull request by brynbodayle**
Fixes lag detailed by issue #7.
Also removes unnecessary recreation of views on tintColorDidChange.

### CODE REVIEW

**Comment by umerkhan-apple** [comment]
@brynbodayle Thanks! This looks great.

r=m3!

File: CareKit/CareCard/OCKHeartView.m [hunk 1]
```diff
@@ -93,12 +123,15 @@ - (void)setValue:(double)value {
 
 - (void)setMaskImage:(UIImage *)maskImage {
     _maskImage = maskImage;
-    [self prepareView];
+    
+    if(maskImage) {
```
Comment (umerkhan-apple) [inline CareKit/CareCard/OCKHeartView.m]: Change `maskImage` to `_maskImage`

File: CareKit/CareCard/OCKHeartView.m [hunk 2]
```diff
@@ -55,35 +55,65 @@ - (void)prepareView {
...
         } completion:^(BOOL finished) {
         }];
     } else {
-        _fillView.frame = CGRectMake(CGRectGetMinX(self.bounds), CGRectGetMaxY(self.bounds), CGRectGetMaxX(self.bounds), -_value * CGRectGetMaxY(self.bounds));
+        [self updateFillViewFrame];
     }
 }
 
+- (void)updateFillViewFrame {
+
```
Comment (umerkhan-apple) [inline CareKit/CareCard/OCKHeartView.m]: Remove extra line.

File: CareKit/CareCard/OCKHeartView.m [hunk 3]
```diff
@@ -55,35 +55,65 @@ - (void)prepareView {
...
     [self insertSubview:_imageView belowSubview:_maskView];
     
-    _fillView = [[UIView alloc] initWithFrame:CGRectMake(CGRectGetMinX(self.bounds), CGRectGetMaxY(self.bounds), CGRectGetMaxX(self.bounds), CGRectGetMaxY(self.bounds))];
+    _fillView = [[UIView alloc] init];
     _fillView.backgroundColor = self.tintColor;
     [self addSubview:_fillView];
 }
 
+- (void)layoutSubviews {
+
```
Comment (umerkhan-apple) [inline CareKit/CareCard/OCKHeartView.m]: Remove extra line.

File: CareKit/CareCard/OCKHeartView.m [hunk 4]
```diff
@@ -55,35 +55,65 @@ - (void)prepareView {
...
+    CGFloat imageWidth = scale * maskImageSize.width;
+    CGFloat imageHeight = scale * maskImageSize.height;
+    
+    _maskView.frame = CGRectMake(0.0, 0.0, imageWidth, imageHeight);
+    _maskView.center = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
+    _maskImageView.frame = _maskView.bounds;
+    _imageView.frame = _maskView.frame;
+    
+    if(_fillView.layer.animationKeys.count == 0) {
+
```
Comment (umerkhan-apple) [inline CareKit/CareCard/OCKHeartView.m]: Remove extra line.

**Comment by umerkhan-apple** [comment]
@brynbodayle friendly ping. :)

**Comment by brynbodayle** [comment]
@umerkhan-apple Ah, done. Thanks for the feedback!

**Comment by umerkhan-apple** [comment]
@brynbodayle Looks great! r=m3 👍

**Pull request merged**